multi-node/aws/pkg/cluster/cluster.go

```diff
@@ -316,24 +316,42 @@ func (c *Cluster) validateDNSConfig(r53 r53Service) error {
 		return nil
 	}
 
-	zonesResp, err := r53.ListHostedZonesByName(&route53.ListHostedZonesByNameInput{
+	listHostedZoneInput := route53.ListHostedZonesByNameInput{
 		DNSName: aws.String(c.HostedZone),
-	})
+	}
+
+	zonesResp, err := r53.ListHostedZonesByName(&listHostedZoneInput)
 	if err != nil {
 		return fmt.Errorf("Error validating HostedZone: %s", err)
 	}
 
 	zones := zonesResp.HostedZones
-	if len(zones) == 0 || (*zones[0].Name != c.HostedZone) {
-		return fmt.Errorf(
-			"HostedZone %s does not exist.  You'll need to create it manually",
-			c.HostedZone,
-		)
+
+	if len(zones) == 0 {
+		return fmt.Errorf("hosted zone %s does not exist", c.HostedZone)
+	}
+
+	var matchingZone *route53.HostedZone = nil
+	for _, zone := range zones {
+		if aws.StringValue(zone.Name) == c.HostedZone {
+			//TODO:(chom) for v1, we will require HostedZoneID be
+			//set, as HostedZone will be going away
```

Comment:
If the plan is to deprecate `HostedZone`, then we should allow a person to specify *only* a `HostedZoneID`.  Looks like this still requires someone to provide `HostedZone`. I would probably expect the steps to be:

1. If `HostedZoneID` is specified: use it (no ambiguity). Fail on no zones found.
1. If only `HostedZone` is specified, fail on len(zones)>1 or no zones found.